src/base/static/components/utils/loader-with-spinner.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import Spinner from "react-spinner";
+
+import { translate } from "react-i18next";
+
+import "./loader-with-spinner.scss";
+
+class LoaderWithSpinner extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      isWithError: false,
+    };
+  }
+
+  componentWillMount() {
+    Promise.all(this.props.promiseData)
```

Comment:
I don't understand why `LoaderWithSpinner` needs to own this state as well as the render prop. I think it is fine to add an `isLoading` state to the StorySidebar, and fetch this promise inside `componentDidMount` of StorySidebar.

Am I missing something? Having this spinner use a render prop seems unnecessarily complicated.

Reply:
I was thinking it would be cleaner for the story sidebar not to be concerned with the Promises. Ideally the sidebar mounts knowing it has all the data it needs to proceed and doesn't need to understand anything before that. That was the point of the loading component. I was imagining this loader would be useful elsewhere (an "atom", as it were).

But maybe that's a YAGNI situation at this point in time?

Reply:
Promises aint no thing. React component lifecycle is built for that. Put them in `componentDidMount`, and add an `async` to allow you to `await` them. ie:

```
async componentDidMount() {
  try {
    const result = await this.myPromise
    console.log(result)
  } catch(e) {
    // do something
  }
}
```
which is equivalent to:

```
componentDidMount() {
  this.myPromise.then(result => {  
    console.log(result)
  }).catch(e => {
    // do something
  })
}
```

Reply:
I see-- that makes sense. Yeah, it's tidier to not have to rely on a separate loading component I suppose.